### DISCUSSION TITLE
Questions about the different branches of Kong

### DISCUSSION THREAD

**Discussion by laurance-moon** [Kong/kong#10241] [Help] [answered]
1. What are the differences between different branches of Kong except for the Openresty version on which it depends?
2. Check whether there is a maintenance plan for each branch, such as the EOL time (planned EOL time) of each branch,evolution strategy of each branch？
thanks.

**Comment by chronolaw** [comment, marked as answer]
This link may be helpful. https://docs.konghq.com/gateway/latest/support-policy/

**Comment by laurance-moon** [comment]
thanks for your reply，it's helpful.
What are the major differences between 2.5.x and 3.1.x? If we want to upgrade from 2.5.x to 3.1.x, do you have an upgrade guide? What adaptation and development are required?

**Reply by chronolaw** [reply]
We have a [blog](https://konghq.com/blog/kong-gateway-3-0) to describe the main features of 3.0, you can get more details in the changelog.md and upgrade.md.

**Comment by laurance-moon** [comment]
ok, thanks.
